orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/tasks/PlanTemplateDependentsTask.java

```diff
@@ -0,0 +1,112 @@
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Component
+public class PlanTemplateDependentsTask implements RetryableTask {
+
+  @Autowired(required = false)
+  private Front50Service front50Service;
+
+  @Autowired
+  private ObjectMapper pipelineTemplateObjectMapper;
+
+  @Autowired
+  private PipelineTemplatePipelinePreprocessor pipelineTemplatePipelinePreprocessor;
+
+  @Nonnull
+  @Override
+  public TaskResult execute(@Nonnull Stage stage) {
+    if (front50Service == null) {
+      throw new UnsupportedOperationException("Front50 is not enabled, no way to fetch pager duty. Fix this by setting front50.enabled: true");
+    }
+
+    if (!stage.getContext().containsKey("pipelineTemplate")) {
+      throw new IllegalArgumentException("Missing required task parameter (pipelineTemplate)");
+    }
+
+    if (!(stage.getContext().get("pipelineTemplate") instanceof String)) {
+      throw new IllegalArgumentException("'pipelineTemplate' context key must be a base64-encoded string: Ensure you're on the most recent version of gate");
+    }
+
+    PipelineTemplate pipelineTemplate = (PipelineTemplate) stage.decodeBase64(
+      "/pipelineTemplate",
+      PipelineTemplate.class,
+      pipelineTemplateObjectMapper
+    );
+
+    List<Map<String, Object>> dependentPipelines = front50Service.getPipelineTemplateDependents(pipelineTemplate.getId());
+
+    Map<String, Object> errorResponses = new HashMap<>();
+    for (Map<String, Object> dependentPipeline : dependentPipelines) {
+      Map<String, Object> request = new HashMap<>();
+      request.put("type", "templatedPipeline");
+      request.put("trigger", new HashMap<>());
+      request.put("config", dependentPipeline.get("config"));
+      request.put("template", pipelineTemplate);
+      request.put("plan", true);
+
+      Map<String, Object> response = pipelineTemplatePipelinePreprocessor.process(request);
+      if (response.containsKey("errors")) {
+        errorResponses.put((String) dependentPipeline.get("id"), response.get("errors"));
+      }
+    }
+
+    Map<String, Object> outputs = new HashMap<>();
+    outputs.put("notification.type", "plantemplatedependents");
+    outputs.put("pipelineTemplate.id", pipelineTemplate.getId());
+    outputs.put("pipelineTemplate.allDependentPipelines", dependentPipelines.stream().map(it -> it.get("id")).collect(Collectors.toList()));
+
+    if (!errorResponses.isEmpty()) {
+      outputs.put("pipelineTemplate.dependentErrors", errorResponses);
+    }
+
+    return new TaskResult(
+      errorResponses.isEmpty() ? ExecutionStatus.SUCCEEDED : ExecutionStatus.TERMINAL,
+      Collections.emptyMap(),
+      outputs
```

Comment:
you expecting these outputs to be global / consumable downstream? I would have expected them to be exported into the current context.

Reply:
or maybe just need to clarify the intent of what `outputs` means ... historically that was for the global context and generally speaking not everything needs to be global.

Reply:
I was a little fuzzy on this - context is definitely what I want.